src/components/user-icon.tsx

```diff
@@ -0,0 +1,83 @@
+import { Avatar, AvatarFallback, AvatarImage } from './ui/avatar';
+import { Button } from './ui/button';
+import Link from 'next/link';
+import type { UserIconInfo } from '../types/user-icon-info';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui/popover';
+import { useState } from 'react';
+import { UserPlus, SquareArrowOutUpRight, UserMinus } from 'lucide-react';
+
+export const UserIcon = ({ iconInfo }: { iconInfo: UserIconInfo }) => {
```

Comment:
iconInfoというオブジェクトにまとめる必要はないです。下の実装で動く
```suggestion
type UserIconProps = {
  id: number;
  image:string;
  is_following:boolean;
}

export const UserIcon = ({ id, image, is_following }: UserIconProps) => {
```